rs-backend/src/load.rs

```diff
@@ -100,29 +109,12 @@ impl InputData {
                 continue;
             }
 
-            /// Parse date from JSON header in file contents.
-            fn parse_from_header(date: &str) -> Result<NaiveDateTime> {
-                // TODO: Determine correct format of header date and move into
-                // variable/constant
-                NaiveDateTime::parse_from_str(date, "%c %z")
-                    .map_err(|err| err.into())
-            }
-
-            /// Parse date from filename.
-            fn parse_from_filename(filename: &str) -> Result<NaiveDateTime> {
-                let date_str = &filename[(filename.find("--").unwrap() + 2)..filename.find(".json").unwrap()];
-
-                // Ignore seconds: they aren't key to the data processing, and
-                // parsing them requires logic to replace them with 0 when
-                // they are absent.
-                NaiveDateTime::parse_from_str(date_str, "%Y-%m-%d-%H-%M")
-                    .map_err(|err| err.into())
-            }
-
             let header = InputHeader {
```

Comment:
Since we're polishing, this struct is actually entirely useless. I think keeping the commit and date as standalone variables makes more sense to me. If the struct is kept then `TestRun::new` should just take the header instead of the date and commit separately.